homeassistant/components/rainbird/switch.py

```diff
@@ -57,49 +69,63 @@ def setup_platform(
             name = zone_config.get(CONF_FRIENDLY_NAME)
             devices.append(
                 RainBirdSwitch(
+                    coordinator,
                     controller,
                     zone,
                     time,
                     name if name else f"Sprinkler {zone}",
                 )
             )
 
-    add_entities(devices, True)
+    try:
+        await coordinator.async_config_entry_first_refresh()
```

Comment:
This was a bit unexpected, but it will work.

Reply:
I feel like i'm missing the best practice. Is there a better way to accomplish the same thing? When I revert this, it seems to create the entities as unavailable w/o scheduling the first attempt.

(Or is this about the exception handling? i was expecting this can get simplified when moving to config entries)

Reply:
Its really not meant to be used in `async_setup_platform`, but since you are planning on converting it to a config entry anyways it seems like a reasonable intermediary step than writing out nearly the same code its using and raising `PlatformNotReady`

Reply:
yeah, ok, that was the reasoning. I'll fast follow and get this cleaned up.